packages/server/src/chat/service.ts

```diff
@@ -0,0 +1,120 @@
+import { Message, uuid } from '@botpress/messaging-base'
+import _ from 'lodash'
+import yn from 'yn'
+import { Service } from '../base/service'
+import { ChannelService } from '../channels/service'
+import { ClientService } from '../clients/service'
+import { ConduitService } from '../conduits/service'
+import { ConfigService } from '../config/service'
+import { ConversationService } from '../conversations/service'
+import { InstanceService } from '../instances/service'
+import { LoggerService } from '../logger/service'
+import { Logger } from '../logger/types'
+import { MappingService } from '../mapping/service'
+import { MessageService } from '../messages/service'
+import { PostService } from '../post/service'
+import { SocketService } from '../socket/service'
+import { WebhookBroadcaster } from '../webhooks/broadcaster'
+import { WebhookService } from '../webhooks/service'
+import { WebhookContent } from '../webhooks/types'
+
+export class ChatService extends Service {
+  private webhookBroadcaster: WebhookBroadcaster
+  private loggingEnabled!: boolean
+  private logger!: Logger
+
+  constructor(
+    private loggers: LoggerService,
+    private configs: ConfigService,
+    private posts: PostService,
+    private channels: ChannelService,
+    private clients: ClientService,
+    private webhooks: WebhookService,
+    private conduits: ConduitService,
+    private conversations: ConversationService,
+    private messages: MessageService,
+    private mappings: MappingService,
+    private instances: InstanceService,
+    private sockets: SocketService
+  ) {
+    super()
+    this.webhookBroadcaster = new WebhookBroadcaster(this.posts, this.webhooks)
+    this.logger = this.loggers.root.sub('chat')
+  }
+
+  async setup() {
+    if (process.env.LOGGING_ENABLED?.length) {
+      this.loggingEnabled = !!yn(process.env.LOGGING_ENABLED)
+    } else if (this.configs.current.logging?.enabled !== null && this.configs.current.logging?.enabled !== undefined) {
+      this.loggingEnabled = !!yn(this.configs.current.logging.enabled)
+    } else {
+      this.loggingEnabled = process.env.NODE_ENV !== 'production'
+    }
```

Comment:
The thing about this solution is that if any of the statements are true then logging is forced to be enabled. In particular with this solution you couldn't turn logging off when it's not in production

Reply:
We would need to have a `||` that considers `false` to be a correct value and only checks the other statements when undefined. We also can't do !!yn because this would override the other settings. So yn should return undefined in the value is undefined, and we only consider other sources of condifugartion if our current setting is at undefined.